packages/medusa/src/services/cart.ts

```diff
@@ -331,19 +331,21 @@ class CartService extends TransactionBaseService {
           rawCart.email = customer.email
         }
 
-        if (!data.region_id) {
+        if (!data.region_id && !data.region) {
```

Comment:
question: Are we not unnecessarily constraining our selves with this validation? If the region id is provided, we will attempt to fetch the region on the lines following this check.

Reply:
here we are checking if none of the id or region is provided. In that case we cannot fetch it without the id

Reply:
You don't have to provide both but only one of them, both are optional but at least one must be provided. as usual, it seems like a little perf improvement but with all the queries accumulated it makes a considerable difference.